cloudfoundry-client-lib/src/test/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImplTest.java

```diff
@@ -1,90 +1,152 @@
-
-        Assert.assertEquals(domains.get("domain.com"), domains.get(uriInfo.get("domainName")));
-        Assert.assertEquals("xyz", uriInfo.get("host"));
-        Assert.assertEquals("/path", uriInfo.get("path"));
+    private void executeWithErrorHandling(Executable executable, Class<? extends RuntimeException> expectedException, String exceptionMessage) throws Throwable {
+        if (expectedException != null) {
+            RuntimeException runtimeException = Assertions.assertThrows(expectedException, executable);
+            Assertions.assertEquals(exceptionMessage, runtimeException.getMessage());
+            return;
+        }
+        executable.execute();
     }
 
-    @Before
-    public void setUpWithEmptyConstructor() throws Exception {
-        controllerClient = new CloudControllerRestClientImpl();
+    private void validateUriInfo(Map<String, String> uriInfo, Map<String, UUID> domainsAsMap, String expectedDomain, String expectedHost,
+                                 String expectedPath) {
+        Assertions.assertEquals(domainsAsMap.get(expectedDomain), domainsAsMap.get(uriInfo.get("domainName")));
+        Assertions.assertEquals(expectedHost, uriInfo.get("host"));
+        Assertions.assertEquals(expectedPath, uriInfo.get("path"));
     }
 
+    private static class Input {
+        private String uri;
+        private List<Domain> domains;
+        private String expectedPath;
+        private String expectedHost;
+        private String expectedDomain;
+        private String errorMessage;
+
+        public String getUri() {
+            return uri;
+        }
+
+        public void setUri(String uri) {
+            this.uri = uri;
+        }
+
+        public List<Domain> getDomains() {
+            return domains;
+        }
+
+        public void setDomains(List<Domain> domains) {
+            this.domains = domains;
+        }
+
+        public String getExpectedPath() {
+            return expectedPath;
+        }
+
+        public void setExpectedPath(String expectedPath) {
+            this.expectedPath = expectedPath;
+        }
+
+        public String getExpectedHost() {
+            return expectedHost;
+        }
+
+        public void setExpectedHost(String expectedHost) {
+            this.expectedHost = expectedHost;
+        }
+
+        public String getExpectedDomain() {
+            return expectedDomain;
+        }
+
+        public void setExpectedDomain(String expectedDomain) {
+            this.expectedDomain = expectedDomain;
+        }
+
+        public String getErrorMessage() {
+            return errorMessage;
+        }
+
+        public void setErrorMessage(String errorMessage) {
+            this.errorMessage = errorMessage;
+        }
+
+        public Map<String, UUID> getDomainsAsMap() {
+            return getDomains().stream()
+                               .collect(Collectors.toMap(Domain::getName, Domain::getUuid));
+        }
+
+        private static class Domain {
```

Comment:
Why not the existing object org.cloudfoundry.client.v2.domains.Domain?

Reply:
CloudDomain used instead